scripts/spack/packages/serac/package.py

```diff
@@ -372,28 +385,36 @@ def initconfig_hardware_entries(self):
         if spec.satisfies("+rocm"):
             entries.append(cmake_cache_option("ENABLE_HIP", True))
 
-            hip_root = spec["hip"].prefix
-            rocm_root = hip_root + "/.."
+            # Add search paths
+            rocm_root = os.path.dirname(spec["llvm-amdgpu"].prefix)
+            entries.append(cmake_cache_path("ROCM_PATH", rocm_root))
+            hip_link_flags = "-L{0}/lib -Wl,-rpath,{0}/lib ".format(rocm_root)
+            hip_link_flags += "-L{0}/llvm/lib -Wl,-rpath,{0}/llvm/lib ".format(rocm_root)
+
+            # Recommended MPI flags
+            hip_link_flags += "-lxpmem "
+            hip_link_flags += "-L/opt/cray/pe/mpich/{0}/gtl/lib ".format(spec["mpi"].version)
+            hip_link_flags += "-Wl,-rpath,/opt/cray/pe/mpich/{0}/gtl/lib ".format(spec["mpi"].version)
+            hip_link_flags += "-lmpi_gtl_hsa "
 
-            # Fix blt_hip getting HIP_CLANG_INCLUDE_PATH-NOTFOUND bad include directory
-            if (self.spec.satisfies('%cce') or self.spec.satisfies('%clang')) and 'toss_4' in self._get_sys_type(spec):
-                # Set the patch version to 0 if not already
-                clang_version= str(self.compiler.version)[:-1] + "0"
-                hip_clang_include_path = rocm_root + "/llvm/lib/clang/" + clang_version + "/include"
-                if os.path.isdir(hip_clang_include_path):
-                    entries.append(cmake_cache_path("HIP_CLANG_INCLUDE_PATH", hip_clang_include_path))
+            if self.is_fortran_compiler("amdflang"):
+                hip_link_flags += "-Wl,--disable-new-dtags "
+                hip_link_flags += "-lflang -lflangrti "
+
+            # Remove extra link library for crayftn
+            if "+fortran" in spec and self.is_fortran_compiler("crayftn"):
+                entries.append(
+                    cmake_cache_string("BLT_CMAKE_IMPLICIT_LINK_LIBRARIES_EXCLUDE", "unwind")
```

Comment:
what does this do?

Reply:
CMake automatically finds libraries under the hood and adds them for you. Sometimes this is helpful and other times it is very much not. This removes this library after CMake adds it.